executor/aggfuncs/aggfunc_test.go

```diff
@@ -243,6 +243,120 @@ func rowMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas
 	return memDeltas, nil
 }
 
+func maxUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
+	memDeltas = make([]int64, srcChk.NumRows())
+	var (
+		preStringVal string
+		preJSONVal   string
+		preEnumVal   types.Enum
+		preSetVal    types.Set
+	)
+
+	for i := 0; i < srcChk.NumRows(); i++ {
+		row := srcChk.GetRow(i)
+		if row.IsNull(0) {
+			continue
+		}
+		switch dataType.Tp {
+		case mysql.TypeString:
+			curVal := row.GetString(0)
+			if i == 0 {
+				memDeltas[i] = int64(len(curVal))
+				preStringVal = curVal
+			} else if curVal > preStringVal {
+				memDeltas[i] = int64(len(curVal)) - int64(len(preStringVal))
+				preStringVal = curVal
+			}
+		case mysql.TypeJSON:
+			curVal := row.GetJSON(0)
+			curStringVal := string(curVal.Value)
+			if i == 0 {
+				memDeltas[i] = int64(len(curStringVal))
+				preJSONVal = curStringVal
+			} else if curStringVal > preJSONVal {
+				memDeltas[i] = int64(len(curStringVal)) - int64(len(preJSONVal))
+				preJSONVal = curStringVal
+			}
+		case mysql.TypeEnum:
+			curVal := row.GetEnum(0)
+			if i == 0 {
+				memDeltas[i] = int64(len(curVal.Name))
+				preEnumVal = curVal
+			} else if curVal.Value > preEnumVal.Value {
+				memDeltas[i] = int64(len(curVal.Name)) - int64(len(preEnumVal.Name))
+				preEnumVal = curVal
+			}
+		case mysql.TypeSet:
+			curVal := row.GetSet(0)
+			if i == 0 {
+				memDeltas[i] = int64(len(curVal.Name))
+				preSetVal = curVal
+			} else if curVal.Value > preSetVal.Value {
+				memDeltas[i] = int64(len(curVal.Name)) - int64(len(preSetVal.Name))
+				preSetVal = curVal
+			}
+		}
+	}
+	return memDeltas, nil
+}
+
+func minUpdateMemDeltaGens(srcChk *chunk.Chunk, dataType *types.FieldType) (memDeltas []int64, err error) {
```

Comment:
Could we  merge the `maxUpdateMemDeltaGens` and the `minUpdateMemDeltaGens` logic into one fucntion?

Reply:
Should we move those `updateMemDeltaGens` to the `func_max_min_test.go`.  They are used by the maxMinAggFunc only.

Reply:
done 😄